Arrowgene.Ddon.Server/Network/PingRequestPacketHandler.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Timers;
+using Arrowgene.Ddon.Shared.Entity;
+
+namespace Arrowgene.Ddon.Server.Network
+{
+    public abstract class PingRequestPacketHandler<TClient, TReqStruct, TResStruct> : RequestPacketHandler<TClient, TReqStruct, TResStruct>
+        where TClient : Client
+        where TReqStruct : class, IPacketStructure, new()
+        where TResStruct : ServerResponse, new()
+    {
+
+        private static readonly double TIMER_INTERVAL_MS = 60000; // 1 minute
+
+        private readonly Timer TimeoutCheckTimer;
+
+        protected PingRequestPacketHandler(DdonServer<TClient> server) : base(server)
+        {
+            TimeoutCheckTimer = new Timer(TIMER_INTERVAL_MS);
+            TimeoutCheckTimer.AutoReset = true;
+            TimeoutCheckTimer.Elapsed += (sender, e) =>
+            {
+                DateTime now = DateTime.UtcNow;
+                foreach (var client in server.ClientLookup.GetAll())
+                {
+                    if ((now - client.PingTime).TotalMilliseconds > TIMER_INTERVAL_MS * 2)
+                    {
+                        // Try messaging the client to ensure it is still alive
+                        // If the client is dead, the send will fail and it'll be cleaned up
+                        var pingRes = BuildPingResponse(client, now);
+                        client.Send(pingRes);
+                    }
+                }
+            };
+            TimeoutCheckTimer.Start();
+
+            // TODO: Find out a good place to dispose TimeoutCheckTimer
```

Comment:
At the very least we should clean this up (as it's a thread-like construct) after the .stop() method is invoked on the CLI program and the servers are requested to be shut down.
Actually, without clean disposal, does this not leave some form of loose thread running as a background process on the system?

Reply:
When the process dies, the OS cleans up the resources associated with it. Disposing is useful to avoid wasting resources during execution.